docs/Using Price Feeds/historical-price-data.html

```diff
@@ -29,19 +29,93 @@
+- `roundId` computed round id. From the above formula, you can think of it as a large number containing `phaseId` and `aggregatorRoundId`.
+
+> 📘 Remark
+>
+> Note the formula above ensures that no matter how many times the underlying aggregator is changed, the proxy `roundId` will always be increasing.
+>
+
+** Example:**
+
+When querying the historical data, it is important to know when we reach the end of the history of the underlying aggregator. Let's illustrate this with an example:
+At the time of writing, `latestRoundData` [function](/docs/price-feeds-api-reference/#latestrounddata) of  [LINK / USD feed on Ethereum Mainnet](/docs/ethereum-addresses/) returns `roundId = 92233720368547771158`. Using this value, we can compute the `phaseId` and `aggregatorRoundId`:
+
+- `phaseId = 92233720368547771158 >> 64` . Right shifting an integer by 64 bits is equivalent to dividing it by 2^64: `phaseId = 92233720368547771158/ 2^64 = 5` . The current phase id is 5 , which means that this proxy has had 5 underlying aggregators since its initial deployment.
+- `aggregatorRoundId = uint64(92233720368547771158)`. This will retrieve the first 64 bits from the right. If you are calculating this off-chain, you can do the following in javascript:
+
+```javascript
+// First parse to BigInt to perform computation with big integers
+const num = BigInt("92233720368547771158");
+const num2 = BigInt("0xFFFFFFFFFFFFFFFF"); // Largest 64bits integer 
+
+console.log(Number(num >> 64n)); // returns 5 (phaseId)
+console.log(Number(num & num2)) // returns 13078 (aggregatorRoundId) . Use & (AND bitwise operator) which sets each bit to _1_ if both bits are _1_
+```
+
+Now that we have the current aggregator's round _13078_ , we can get its historical data by looping over `getRoundData` [function](/docs/price-feeds-api-reference/#getrounddata) :
+- starting from the first round _92233720368547758081_ (result of _92233720368547771158 - 13078 + 1_).
+- until the current round: _92233720368547771158_. 
+
+_Note_: if we want the historical data for previous aggregators, we have to decrement the `phaseId` and start from round _1_. For phase _4_, we can get the starting roundId off-chain in javascript : 
+
+```javascript
+const phaseId = BigInt("4");
+const aggregatorRoundId = BigInt("1");
+
+roundId = (phaseId << 64n) | aggregatorRoundId // returns 73786976294838206465n
+```
+
+We can loop over the `getRoundData` [function](/docs/price-feeds-api-reference/#getrounddata) , starting at _73786976294838206465_ and incrementing it until we get a revert. This means that we have reached the last round for the underlying aggregator. The same process could be repeated for previous `phaseIds` (3,2,1).
+
+> 🚧 Looping on-chain
+>
+> The examples showed how to loop off-chain to fetch all historical data from a given proxy. Technically, you could also write a similar code on-chain but please be aware that this could cause **very high gas prices** if a state is changed within the same function.
```

Comment:
```suggestion
> The examples showed how to loop off-chain to fetch all historical data from a given proxy. You could also write a similar code on-chain, but be aware that this could cause **very high gas prices** if a state is changed within the same function.
```